pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -615,26 +621,47 @@ func (m *kubeGenericRuntimeManager) computePodResizeAction(pod *v1.Pod, containe
 			currentContainerResources: &currentResources,
 		}
 		// Order the container updates such that resource decreases are applied before increases
-		switch {
-		case specValue > statusValue: // append
-			changes.ContainersToUpdate[rName] = append(changes.ContainersToUpdate[rName], cUpdateInfo)
-		case specValue < statusValue: // prepend
-			changes.ContainersToUpdate[rName] = append(changes.ContainersToUpdate[rName], containerToUpdateInfo{})
-			copy(changes.ContainersToUpdate[rName][1:], changes.ContainersToUpdate[rName])
-			changes.ContainersToUpdate[rName][0] = cUpdateInfo
+		if isRestartableInitContainer {
+			switch {
+			case specValue > statusValue: // append
+				changes.InitContainersToUpdate[rName] = append(changes.InitContainersToUpdate[rName], cUpdateInfo)
+			case specValue < statusValue: // prepend
+				changes.InitContainersToUpdate[rName] = append(changes.InitContainersToUpdate[rName], containerToUpdateInfo{})
+				copy(changes.InitContainersToUpdate[rName][1:], changes.InitContainersToUpdate[rName])
+				changes.InitContainersToUpdate[rName][0] = cUpdateInfo
+			}
+		} else {
+			switch {
+			case specValue > statusValue: // append
+				changes.ContainersToUpdate[rName] = append(changes.ContainersToUpdate[rName], cUpdateInfo)
+			case specValue < statusValue: // prepend
+				changes.ContainersToUpdate[rName] = append(changes.ContainersToUpdate[rName], containerToUpdateInfo{})
+				copy(changes.ContainersToUpdate[rName][1:], changes.ContainersToUpdate[rName])
+				changes.ContainersToUpdate[rName][0] = cUpdateInfo
+			}
 		}
```

Comment:
Yeah it should work since `containersToUpdate` is a map and maps are always passed/assigned by reference in Go. Made the change as per your suggestion. Thanks!